akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,23 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.foreach {
```

Comment:
Thanks for pointing this out, it seemed to work, so I just took it for granted without looking further into (my initial thoughts were that the iterator contains the names of the  children already resolved by `ActorSelection.deliverSelection` for example in case of wildcard elements in the path). I'll fix it soon